tests/core/job/test_job_manager.py

```diff
@@ -150,13 +152,94 @@ def test_force_deleting_unfinished_job():
     _Scheduler._update_job_config()
     task = _create_task(inner_lock_multiply, name="delete_unfinished_job")
     with lock:
-        job = _Scheduler.submit_task(task)
+        job = _Scheduler.submit_task(task, "submit_id")
         with pytest.raises(JobNotDeletedException):
             _JobManager._delete(job, force=False)
         _JobManager._delete(job, force=True)
     assert _JobManager._get(job.id) is None
 
 
+def test_cancel_single_job():
+    Config.configure_job_executions(mode=JobConfig._STANDALONE_MODE, nb_of_workers=2)
+    _Scheduler._update_job_config()
+    task = _create_task(inner_lock_multiply, name="delete_unfinished_job")
+
+    with lock:
+        job = _Scheduler.submit_task(task, "submit_id")
+
+        assert job.is_running()
+        assert len(_Scheduler._processes) == 1
+        _JobManager._cancel(job.id)
+        assert job.is_cancelled()
+        assert len(_Scheduler._processes) == 0
+    assert job.is_cancelled()
```

Comment:
I would like to tests that when killing a process, the nb of available workers is updated as well.

```suggestion
    assert _Scheduler._dispatcher._nb_available_workers == 2
    with lock:
        job = _Scheduler.submit_task(task, "submit_id")

        assert job.is_running()
        assert len(_Scheduler._processes) == 1
        assert _Scheduler._dispatcher._nb_available_workers == 1
        _JobManager._cancel(job.id)
        assert job.is_cancelled()
        assert len(_Scheduler._processes) == 0
    assert job.is_cancelled()
    assert _Scheduler._dispatcher._nb_available_workers == 2
```

Reply:
@jrobinAV Oh we aren't killing the process at the moment. But yes, I agree we should have that when we implement the killing process. I'll take note of that to make sure it is included next time

Reply:
Yes, right, but still we can still test now, no?

Reply:
I'd say we can. Just want to stress that the flow would be a bit different, what I mean by that is after `with lock:` the process dispatched by submitting the task will be completed and it should update the nb_available_workers to 2. Anyway, it certainly is possible, I will add it to the test :D